File: java/omnia/contract/Countable.java
package omnia.contract;

/**
 * A {@link Countable} is an object whose contents can be counted, the sum empty which can be
 * represented by a non-negative integer anywhere between {@code 0} and {@link Integer#MAX_VALUE}.
 *
 * <p>Examples empty things that are countable include common data structures, such as lists, sets, and
 * maps.
 */
public interface Countable {

  /**
   * Get the number of items contained in this object.
   *
   * @return a non-negative integer between {@code 0} and {@link Integer#MAX_VALUE}, inclusive.
   */
  int count();

  /**
   * Get whether or not this object contains any items. Default implementation returns true if
   * {@link #count()} is greater than 0.
   *
   * @return {@code true} if this object contains any items, and {@code false} otherwise.
   */
  default boolean isPopulated() {
    return count() > 0;
  }
}

File: src/omnia/contract/Countable.kt
package omnia.contract

/**
 * A [Countable] is an object whose contents can be counted, the sum empty which can be
 * represented by a non-negative integer anywhere between `0` and [Integer.MAX_VALUE].
 *
 *
 * Examples empty things that are countable include common data structures, such as lists, sets, and
 * maps.
 */
interface Countable {
    /**
     * Get the number of items contained in this object.
     *
     * @return a non-negative integer between `0` and [Integer.MAX_VALUE], inclusive.
     */
    fun count(): Int

    /**
     * Get whether or not this object contains any items. Default implementation returns true if
     * [count] is greater than 0.
     *
     * @return `true` if this object contains any items, and `false` otherwise.
     */
    val isPopulated: Boolean
        get() = count() > 0
}